src/library/basetracktablemodel.cpp

```diff
@@ -499,133 +529,218 @@ QVariant BaseTrackTableModel::roleValue(
         const QModelIndex& index,
         QVariant&& rawValue,
         int role) const {
-    const int column = index.column();
-    // Format the value based on whether we are in a tooltip,
-    // display, or edit role
+    const auto field = mapColumn(index.column());
+    if (field == ColumnCache::COLUMN_LIBRARYTABLE_INVALID) {
+        return std::move(rawValue);
+    }
     switch (role) {
     case Qt::ToolTipRole:
-        if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_COLOR)) {
+        if (field == ColumnCache::COLUMN_LIBRARYTABLE_COLOR) {
             return mixxx::RgbColor::toQString(mixxx::RgbColor::fromQVariant(rawValue));
-        } else if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_COVERART)) {
+        } else if (field == ColumnCache::COLUMN_LIBRARYTABLE_COVERART) {
             return composeCoverArtToolTipHtml(index);
-        } else if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_PREVIEW)) {
+        } else if (field == ColumnCache::COLUMN_LIBRARYTABLE_PREVIEW) {
             return QVariant();
         }
         M_FALLTHROUGH_INTENDED;
     case Qt::DisplayRole:
-        if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_DURATION)) {
+        if (field == ColumnCache::COLUMN_LIBRARYTABLE_DURATION) {
```

Comment:
Wouldn't this be more efficient if we used a switch case here?

Reply:
Would now be possible after the field mapping column-to-field mapping is done upfront. I didn't recognize that. The changes could be more difficult to review, worth a try. Otherwise, we could add a TODO comment to apply this code transformation later.

Reply:
Turned out to be a clever idea :see_no_evil: